.github/workflows/ci.yml

```diff
@@ -90,37 +75,53 @@ jobs:
       env:
         COLOR: 'yes'
       run: |
-        make test-only
+        make test
         python -m coverage xml
     - name: Upload coverage
       uses: codecov/codecov-action@v3
       with:
         file: ./coverage.xml
         flags: unit
-        fail_ci_if_error: false
+
+  check:  # This job does nothing and is only used for the branch protection
+    if: always()
+
+    needs: [lint, test]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
+      with:
+        jobs: ${{ toJSON(needs) }}
 
   deploy:
     name: Deploy
+    environment: release
```

Comment:
@Dreamsorcerer let's migrate to an environment called `pypi` instead of `release`. This allows us to start having continuous TestPyPI publishing smoke healthcheck by introducing another env — `testpypi`. Also, we can start injecting the URL param there too. See: https://github.com/aio-libs/aiomysql/blob/4909ba1/.github/workflows/ci-cd.yml#L649-L654